### PR TITLE
Fix bugs found in PE Dry Run

### LINKED ISSUE #144: [PE-D] Invalid Index
In the case where 1 invalid and 1 valid index is entered for the Plan feature, the program still says that the index entered is invalid without registering that there is 1 valid index entered.

-------------
Labels: `severity.VeryLow` `type.FeatureFlaw`
original: nigellenl/ped#6

### LINKED ISSUE #146: [PE-D] No Headers for List
Makes it difficult for new users to identify the columns (especially the Priority).

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: nigellenl/ped#4

### LINKED ISSUE #152: [PE-D] show-plan command doesn't throw any exceptions when there are no plans
As mentioned in the title, the show-plan command does not throw any exceptions when there are no plans. Instead, the app just sits there waiting for the next user input.

This can be recreated by calling the "show-plan" command when there are no study plans stored.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: JeremiasLiew/ped#14

### LINKED ISSUE #154: [PE-D] Study plan logging
Study plan logging does not seem to be logging anything, even though I already created a study plan and tried every single command listed in the user guide

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: JeremiasLiew/ped#12

### LINKED ISSUE #156: [PE-D] FAQ for User Guide doesn't have a proper answer
FAQ answer does not seem correct even though the question is a legit one.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: JeremiasLiew/ped#10

### LINKED ISSUE #159: [PE-D] Unused notation in User Guide
The user guide states that "Items with ...​ after them can be used multiple times including zero times e.g. [DETAILS]...​ can be omitted or used as Detail_1 Detail_2."

While the number of details that can be entered are indeed variable, this "..." notation is not used anywhere in the userguide or the help page.

e.g. for flashcards, no "..." is used with ```DETAIL```

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: JeremiasLiew/ped#7

### LINKED ISSUE #163: [PE-D] Cluttered UI
A lack of spacing between outputs causes the ui to be very cluttered and hard to read. This can be recreated by using any commands that are not creating a new person/event/other flashcard, as the create flashcard commands are the only ones with a space/empty line after the output.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: JeremiasLiew/ped#3

### LINKED ISSUE #165: [PE-D] How to stop entering details not specified in User Guide
Look like in order to stop entering new details and create the event/person/other flashcard, the user needs to enter a blank input. However, this is not mentioned anywhere in the user guide, which does not explain how to stop entering new details.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: JeremiasLiew/ped#1

### LINKED ISSUE #168: Remove unused Ui methods

### PR DESCRIPTION
**Pull request by Zhilin-Huang**

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -47,7 +47,8 @@ The application will then prompt the user to enter the following fields:
 - `SUMMARY`: can be in a natural language format
 - `DETAIL`: can be in a natural language format
 
-There can be 0 or multiple `DETAIL` added for an Event Flashcard.
+There can be 0 or multiple `DETAIL` added for an Event Flashcard. To stop entering details, just enter an empty line
+by hitting `RETURN`.
```
Comment (g-lilian) [inline docs/UserGuide.md]: `ENTER`/`RETURN`? It's a small issue but on non-Macs the key is called Enter :)

File: docs/UserGuide.md [hunk 2]
```diff
@@ -293,9 +293,9 @@ Format:
 
 ## FAQ
 
-**Q**: How do I transfer my data to another computer? 
+**Q**: Does History Flashcard have GUI version?
 
-**A**: Well, write the User Guide in active voice anyway.
+**A**: Not at this moment. But we are working on it - keep a lookout for our new releases!
```
Comment (g-lilian) [inline docs/UserGuide.md]: Haha, but I don't think a GUI will ever happen??

**Review by g-lilian — CHANGES_REQUESTED** [review]
Generally lgtm! Thanks for fixing all these

File: docs/UserGuide.md (hunk 1 above)
Comment (Zhilin-Huang) [reply docs/UserGuide.md]: Oops I'll update haha

File: docs/UserGuide.md (hunk 2 above)
Comment (Zhilin-Huang) [reply docs/UserGuide.md]: I cannot think of other things to add in FAQ lol So I just add some random stuff. New release can be v3.0 which we'll never work on hahahahah

Comment (g-lilian) [reply docs/UserGuide.md]: Hmm one possible FAQ:
Where can I find the files containing all my flashcard information?
In folder xxx, the file xxx.txt stores the flashcards in the following format: ...
It's info that doesn't really fall under any of the commands, but some users may want to know.

Comment (Zhilin-Huang) [reply docs/UserGuide.md]: > Hmm one possible FAQ:
> Where can I find the files containing all my flashcard information?
> In folder xxx, the file xxx.txt stores the flashcards in the following format: ...
> It's info that doesn't really fall under any of the commands, but some users may want to know.

Sounds good, I'll update :)

**Review by g-lilian — APPROVED** [review]
Nice :)

**Pull request merged**